src/main/java/seedu/address/logic/parser/FindExpressionParser.java

```diff
@@ -0,0 +1,288 @@
+        } else {
+            Token token = consume(Token.Type.CONDITION);
+            // check if text contains a slash and is a valid condition
+            if (!token.text.contains("/") || token.text.startsWith("/") || token.text.endsWith("/")) {
+                throw new ParseException("Invalid condition: " + token.text);
+            }
+            // split by slash but include slash in substrings
+            String[] parts = token.text.split("(?<=/)");
+            FindSupportedField field = FindSupportedField.fromPrefix(parts[0].trim().toLowerCase());
+            return new ConditionNode(field, parts[1].trim());
+        }
+    }
+
+    private boolean match(Token.Type type) {
+        return check(type);
+    }
+
+    private boolean check(Token.Type type) {
+        if (isAtEnd()) {
+            return false;
+        }
+        return peek().type == type;
+    }
+
+    private Token consume(Token.Type type) throws ParseException {
+        if (check(type)) {
+            return tokens.get(pos++);
+        }
+        if (isAtEnd()) {
+            throw new ParseException("Find command received an invalid filter string: "
+                    + "Expected token of type " + type + " but reached end of input!");
+        }
+        throw new ParseException("Find command received an invalid filter string: "
+                + "Expected token of type " + type + " but found " + peek().type);
+    }
+
+    private Token peek() {
+        return tokens.get(pos);
+    }
+
+    private boolean isAtEnd() {
+        return pos == tokens.size();
+    }
+
+    /**
+     * Enum representing the logical operators supported by the parser.
+     */
+    enum FindExpressionOperator {
+        AND, OR
+    }
+
+    /**
+     * Enum representing the fields that can be searched.
+     */
+    enum FindSupportedField {
+
+        NAME(PREFIX_NAME.getPrefix()),
+        PHONE(PREFIX_PHONE.getPrefix()),
+        EMAIL(PREFIX_EMAIL.getPrefix()),
+        ADDRESS(PREFIX_ADDRESS.getPrefix()),
+        TAG(PREFIX_TAG.getPrefix());
+
+        private final String prefix;
+
+        FindSupportedField(String prefix) {
+            this.prefix = prefix;
+        }
+
+        /**
+         * Returns the corresponding field for a given prefix.
+         *
+         * @param prefix The prefix to look up.
+         * @return The supported field corresponding to the given prefix.
+         * @throws ParseException if the prefix is not supported.
+         */
+        public static FindSupportedField fromPrefix(String prefix) throws ParseException {
+            for (FindSupportedField field : FindSupportedField.values()) {
+                if (field.prefix.equals(prefix)) {
+                    return field;
+                }
+            }
+            throw new ParseException("No supported field with prefix " + prefix);
+        }
+    }
+
+    /**
+     * Abstract class representing a node in the expression tree.
+     */
+    abstract static class ExprNode {
```

Comment:
If we switch to `parseToPredicate` this can be made private

Reply:
Makes sense, fixed in c1b642af